app/src/main/java/de/danoeh/antennapod/discovery/ItunesTopListLoader.java

```diff
@@ -23,24 +24,45 @@
 import java.util.Locale;
 import java.util.concurrent.TimeUnit;
 
+import static android.content.Context.MODE_PRIVATE;
+
 public class ItunesTopListLoader {
     private static final String TAG = "ITunesTopListLoader";
     private final Context context;
+    public static final String PREF_KEY_COUNTRY_CODE = "country_code";
+    public static final String PREFS = "CountryRegionPrefs";
+    public static final String DISCOVER_HIDE_FAKE_COUNTRY_CODE = "00";
+    public static final String COUNTRY_CODE_UNSET = "99";
 
     public ItunesTopListLoader(Context context) {
         this.context = context;
     }
 
-    public Single<List<PodcastSearchResult>> loadToplist(int limit) {
+    public Single<List<PodcastSearchResult>> loadToplist() {
+        String defaultCountry = Locale.getDefault().getCountry();
+        SharedPreferences prefs = context.getSharedPreferences(PREFS, MODE_PRIVATE);
+        String countryCode = prefs.getString(PREF_KEY_COUNTRY_CODE, COUNTRY_CODE_UNSET);
+        return this.loadToplist(countryCode, 25);
+    }
+
+    public Single<List<PodcastSearchResult>> loadToplist(String country, int limit) {
         return Single.create((SingleOnSubscribe<List<PodcastSearchResult>>) emitter -> {
-            String country = Locale.getDefault().getCountry();
             OkHttpClient client = AntennapodHttpClient.getHttpClient();
-            String feedString;
+            String feedString = "{}";
+            String loadCountry = country;
+            if (COUNTRY_CODE_UNSET.equals(country)) {
+                loadCountry = Locale.getDefault().getCountry();
+            }
             try {
-                feedString = getTopListFeed(client, country, limit);
+                feedString = getTopListFeed(client, loadCountry, limit);
             } catch (IOException e) {
-                feedString = getTopListFeed(client, "us", limit);
+                if (COUNTRY_CODE_UNSET.equals(country)) {
+                    feedString = getTopListFeed(client, "US", limit);
+                } else {
+                    emitter.onError(e);
```

Comment:
I think there should be a `return` call. Otherwise, the program continues to parse the non-existing feed.

Reply:
🐇 done